pkg/store/pipelineconfig/store.go

```diff
@@ -0,0 +1,238 @@
+func NewAtomixStore(client primitive.Client) (Store, error) {
+	pipelineConfigsAtomicMap, err := atomixclient.AtomicMap[p4rtapi.PipelineConfigID, *p4rtapi.PipelineConfig](client)("device-provisioner-pipeline-configurations").
+		Codec(generic.GoGoProto[*p4rtapi.PipelineConfig](&p4rtapi.PipelineConfig{})).
+		Get(context.Background())
+
+	if err != nil {
+		return nil, errors.FromAtomix(err)
+	}
+
+	store := &configurationStore{
+		pipelineConfigs: pipelineConfigsAtomicMap,
+	}
+
+	return store, nil
+
+}
+
+func (s *configurationStore) Get(ctx context.Context, id p4rtapi.PipelineConfigID) (*p4rtapi.PipelineConfig, error) {
+	// If the pipeline config is not already in the cache, get it from the underlying primitive.
+	entry, err := s.pipelineConfigs.Get(ctx, id)
+	if err != nil {
+		return nil, errors.FromAtomix(err)
+	}
+
+	// Decode and return the Configuration.
+	configuration := entry.Value
+	if err := decodePipelineConfiguration(entry, configuration); err != nil {
+		return nil, errors.NewInvalid("pipeline config decoding failed: %v", err)
+	}
+	return entry.Value, nil
+}
+
+func (s *configurationStore) Create(ctx context.Context, pipelineConfig *p4rtapi.PipelineConfig) error {
+	if pipelineConfig.ID == "" {
+		return errors.NewInvalid("no pipeline pipelineconfig ID specified")
+	}
+	if pipelineConfig.TargetID == "" {
+		return errors.NewInvalid("no target ID specified")
+	}
+	if pipelineConfig.Revision != 0 {
+		return errors.NewInvalid("cannot create pipeline pipeline config with revision")
+	}
+	if pipelineConfig.Version != 0 {
+		return errors.NewInvalid("cannot create pipeline pipeline config with version")
+	}
+	pipelineConfig.Revision = 1
+	pipelineConfig.Created = time.Now()
+	pipelineConfig.Updated = time.Now()
+	// Create the entry in the underlying map primitive.
+	entry, err := s.pipelineConfigs.Put(ctx, pipelineConfig.ID, pipelineConfig)
+	if err != nil {
+		return errors.FromAtomix(err)
+	}
+
+	// Decode the pipeline config from the returned entry.
+	if err := decodePipelineConfiguration(entry, pipelineConfig); err != nil {
+		return errors.NewInvalid("pipelineConfig decoding failed: %v", err)
+	}
+	return nil
+}
+
+func (s *configurationStore) Update(ctx context.Context, pipelineConfig *p4rtapi.PipelineConfig) error {
+	if pipelineConfig.ID == "" {
+		return errors.NewInvalid("no pipelineConfig ID specified")
+	}
+	if pipelineConfig.TargetID == "" {
+		return errors.NewInvalid("no target ID specified")
+	}
+	if pipelineConfig.Revision == 0 {
+		return errors.NewInvalid("pipelineConfig must contain a revision on update")
+	}
+	if pipelineConfig.Version == 0 {
+		return errors.NewInvalid("pipelineconfig must contain a version on update")
+	}
+	pipelineConfig.Revision++
+	pipelineConfig.Updated = time.Now()
+
+	// Update the entry in the underlying map primitive using the pipelineconfig version
+	// as an optimistic lock.
+	entry, err := s.pipelineConfigs.Put(ctx, pipelineConfig.ID, pipelineConfig)
```

Comment:
Similarly, you should use the `Update` method here to avoid creating a new entry when `Update` is called, and we also need to be using optimistic locking to avoid data loss from concurrent updates.